### DISCUSSION TITLE
关于mosdns向上游surge请求解析的问题

### DISCUSSION THREAD

**Discussion by lingkai995** [IrineSistiana/mosdns#677] [General]
forward_remote使用的surge的198.18.0.2，期望是能从198.18.0.2获取到surge给到的fake ip。但是，目前的情况是：从tcp请求mosdns10.10.10.20解析能正常返回fake ip，然而从udp请求却是真实ip。

![image](https://github.com/IrineSistiana/mosdns/assets/15700604/d9c46793-eaff-465f-9fbb-c178d84488b3)

小白苦恼这问题两天了，求帮助~

配置文件如下，请大佬帮忙审审
`log:
  level: debug

plugins:
  # 缓存
  - tag: cache
    type: cache
    args:
      size: 10240
      lazy_cache_ttl: 86400
      
  # 转发至国内DNS，并发查询
  - tag: forward_local
    type: forward
    args:
      concurrent: 3
      upstreams:
        - addr: udp://61.139.2.69
        - addr: udp://119.29.29.29
        - addr: udp://223.5.5.5

  # 转发至国外DNS，并发查询
  - tag: forward_remote
    type: forward
    args:
      concurrent: 1
      upstreams:
        - addr: udp://198.18.0.2

  - tag: to_remote
    type: sequence
    args:
      - exec: $forward_remote
      #- matches: "!resp_ip 198.18.0.0/16"
      #  exec: drop_resp
      #- exec: accept

  # 主运行序列
  - tag: main_sequence
    type: sequence
    args:
    # 拒绝TYPE65
      - matches: qtype 65
        exec: reject 3

    # 在reject-list里的reject
      - matches: qname &/etc/mosdns/reject-list.txt &/etc/mosdns/my-reject-list.txt
        exec: reject
    
    # 所有请求都优先返回ipv4，屏蔽ipv6  
      - exec: prefer_ipv4
        
    # 在direct-list里的转发到local dns
      - matches: qname &/etc/mosdns/direct-list.txt
        exec: $forward_local

        
    # 在proxy-list里的转发到remote dns
      - matches: qname &/etc/mosdns/proxy-list.txt
        exec: $to_remote

  # 启动监听服务
  - tag: udp_server
    type: udp_server
    args:
      entry: main_sequence
      listen: 0.0.0.0:53

  - tag: tcp_server
    type: tcp_server
    args:
      entry: main_sequence
      listen: 0.0.0.0:53
      idle_timeout: 10
`

**Comment by Wu-xxxx** [comment]
老哥，你这个问题解决了吗
